### DISCUSSION TITLE
microprofile-openapi dependency version

### DISCUSSION THREAD

**Discussion by ottuzzi-at-vegans** [quarkusio/quarkus#31341] [Q&A] [answered]
Hi,

   I'm using quarkus 2.16.3.Final and I have some doubt on dependencies in my project.
This is the dependency tree fragment that puzzles me:

```
[INFO] |  |  +- io.quarkus:quarkus-smallrye-openapi:jar:2.16.3.Final:compile
[INFO] |  |  |  +- io.smallrye:smallrye-open-api-core:jar:3.1.1:compile
[INFO] |  |  |  |  +- org.eclipse.microprofile.openapi:microprofile-openapi-api:jar:2.0.1:compile
[INFO] |  |  |  |  +- org.eclipse.microprofile.config:microprofile-config-api:jar:2.0.1:compile
[INFO] |  |  |  |  \- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.14.2:compile
[INFO] |  |  |  |     \- org.yaml:snakeyaml:jar:1.33:compile
```

If I look at the dependency of **io.smallrye:smallrye-open-api-core**[1] I see it depends on **org.eclipse.microprofile.openapi:microprofile-openapi-api 3.1** as it is confirmed also from eclipse:

![image](https://user-images.githubusercontent.com/104214595/220641610-abf950a6-f317-422d-80f1-bc5fb8825de7.png)

Can anyone help me to understand what is happening here?

Thanks in advance
Piero

[1]https://github.com/smallrye/smallrye-open-api/blob/3.1.1/pom.xml

[a comment by an automated account is not shown]

**Comment by MikeEdgar** [comment, marked as answer]
@ottuzzi-at-vegans , smallrye-open-api 3.1 and above uses MicroProfile OpenAPI 3.1, but _supports_ 2.0 and 3.0 as well. Until just recently Quarkus managed the version of the MicroProfile dependency to 2.0.1. The 3.0 release of Quarkus will have what you expect.

**Reply by ottuzzi-at-vegans** [reply]
Thank you @MikeEdgar ... Now I understand what I was missing. 

Thank you
